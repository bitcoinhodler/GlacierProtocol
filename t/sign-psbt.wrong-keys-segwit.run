#!/bin/bash

# Test validation of extra keys. 3rd key does not belong to this address.



$GLACIERSCRIPT --regtest=$1 sign-psbt << INPUT 2>&1
y
y
y
y
y
y
../../t/sign-psbt.wrong-keys-segwit.psbt
y
4
cUY5DRXSmpqT7WpBAkjFz6PPxHoPJ2Cw2VNCa7muihpRjAUJVycs
cQTQEE1ooQHpVi5C4aTcbYPSUMzY2gF1DVdafUsEwnYYaT97UUMy
cPSsBu9SyNVAS2Evy3m4ELFx7KGnudH3N77Es83nafa2xVWJGRSe
INPUT

retVal=$?
if [ $retVal -eq 0 ]; then
    echo "Error: expected glacierscript.py to fail"
    exit 1
fi
