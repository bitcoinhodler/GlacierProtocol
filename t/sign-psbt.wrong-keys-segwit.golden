Are you running this on a computer WITHOUT a network connection of any kind? (y/n)?Have the wireless cards in this computer been physically removed? (y/n)?Are you running on battery power? (y/n)?Are you running on an operating system booted from a USB drive? (y/n)?Is your screen hidden from view of windows, cameras, and other people? (y/n)?Are smartphones and all other nearby devices turned off and in a Faraday bag? (y/n)?Input a filename located in the current directory which contains the PSBT:
0.02000000 BTC in unspent inputs from cold storage address 2Mv6fmwU6Nj6tiSNHzxSDz8vFiEJkYQ46Fm
0.01998100 BTC going to destination address 2NBNqB9wqYYxgcCSeqNrR9RmHc4NsLm7jWe
Fee amount: 0.00001900
Confirm? (y/n): 
How many private keys will you be signing this transaction with (at least 2 required to complete transaction)?
#: Key #1: Key #2: Key #3: ERROR: that key does not belong to this source address. Exiting...
